pulsar-broker/src/test/java/com/yahoo/pulsar/broker/service/ReplicatorTest.java

```diff
@@ -671,6 +676,70 @@ public void testReplicatorProducerClosing() throws Exception {
         assertNull(producer);
     }
 
+    /**
+     * Issue #199
+     *
+     * It verifies that: if the remote cluster reaches backlog quota limit, replicator temporarily stops and once the
+     * backlog drains it should resume replication.
+     *
+     * @throws Exception
+     */
+
+    @Test(enabled = true, priority = -1)
+    public void testResumptionAfterBacklogRelaxed() throws Exception {
+
+        List<RetentionPolicy> policies = Lists.newArrayList();
+        policies.add(RetentionPolicy.producer_exception);
+        policies.add(RetentionPolicy.producer_request_hold);
+
+        for (RetentionPolicy policy : policies) {
+
+            DestinationName dest = DestinationName.get(String.format("persistent://pulsar/global/ns1/%s", policy));
+
+            // Producer on r1
+            MessageProducer producer1 = new MessageProducer(url1, dest);
+
+            // Consumer on r1
+            MessageConsumer consumer1 = new MessageConsumer(url1, dest);
+
+            // Consumer on r2
+            MessageConsumer consumer2 = new MessageConsumer(url2, dest);
+
+            // Replicator for r1 -> r2
+            PersistentTopic topic = (PersistentTopic) pulsar1.getBrokerService().getTopicReference(dest.toString());
+            PersistentReplicator replicator = topic.getPersistentReplicator("r2");
+
+            // Produce 10 messages to r1, and it will be replicated to r2
+            producer1.produce(10);
+            consumer1.receive(10);
+
+            // Restrict backlog quota limit to 1
+            admin1.namespaces().setBacklogQuota("pulsar/global/ns1", new BacklogQuota(1, policy));
```

Comment:
Can we setup the test this way? If you want, you can create another test.
1. limit quota to 1 message
2. Publish 10 messages
3. verify receive times out after first message is received(since quota is full)
4. Increase quota to 10
5. Verify we receive the remaining 9 messages.